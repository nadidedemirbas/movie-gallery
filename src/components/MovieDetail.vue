<template>
  <div class="detail-main">
    <div>
      <img class="image-fluid detail-image" :src="movie.Poster" :alt="title">
    </div>
    <div>
      <div class="detail-title">
        {{ movie.Title }}
      </div>
      <div class="detail-plot">
        Plot: {{ movie.Plot }}
      </div>
      <div class="detail-more">
        Released: {{ movie.Released }} <br/>
        Rating: {{ movie.imdbRating }} <br/>
        Genre: {{ movie.Genre }} <br/>
        Awards: {{ movie.Awards }}
      </div>
    </div>

  </div>
</template>

<script>
export default {
  name: "MovieDetail",
  data() {
    return {
      movie: {}
    }
  },
  methods: {
    async getMovieDetail(id) {
      fetch('https://fake-movie-api.onrender.com/api/movies', {
        headers: {
          AuthKey: 'RmFrZU1vdmllQVBJLWM5MzU1ZjE0LTM3ZGUtNDAzNC1hMTAzLTRkOTcxOTYwOWRmNg=='
        }
      })
      .then((res) => res.json())
      .then((res) => {
        console.log(res)
        this.movie = res.find(movie => movie.imdbID === id);
      })
    }
  },
  mounted() {
    const id = this.$route.params.id;
    this.getMovieDetail(id);
  }
}
</script>

<style scoped>

</style>