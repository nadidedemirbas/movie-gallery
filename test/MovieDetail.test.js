import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'

import MovieDetail from '../src/components/MovieDetail.vue'

const mockMovie = {
    Title: 'mock-title',
    Poster: 'mock-poster',
    Plot: 'mock-plot',
    Released: 'mock-release-date',
    imdbRating: 'mock-rating',
    Genre: 'mock-genre',
    Awards: 'mock-awards'
}

const mockRoute = {
    params: {
        id: 1
    }
}

describe('MovieDetail Component', () => {
    const wrapper = mount(MovieDetail, {
        data() {
            return {
                movie: mockMovie
            }
        },
        global: {
          mocks: {
            $route: mockRoute
          }
        }
    });

    it('should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    it('should show movie title correctly', () => {
        const title = wrapper.find('.detail-title').text();
        expect(title).toBe('mock-title');
    })

    it('should show plot correctly', () => {
        const plot = wrapper.find('.detail-plot').text();
        expect(plot).toBe('Plot: mock-plot')
    })

    it('should show more details', () => {
        const moreDetailsWrapper = wrapper.find('.detail-more').text();
        expect(moreDetailsWrapper).toContain('mock-rating')
        expect(moreDetailsWrapper).toContain('mock-genre')
        expect(moreDetailsWrapper).toContain('mock-awards')
        expect(moreDetailsWrapper).toContain('mock-release-date')
    })
});