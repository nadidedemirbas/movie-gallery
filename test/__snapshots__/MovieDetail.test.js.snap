// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieDetail Component > should render correctly 1`] = `
"<div class=\\"detail-main\\">
  <div><img class=\\"image-fluid detail-image\\" src=\\"mock-poster\\" alt=\\"mock-title\\"></div>
  <div>
    <div class=\\"detail-title\\">mock-title</div>
    <div class=\\"detail-plot\\"> Plot: mock-plot</div>
    <div class=\\"detail-more\\"> Released: mock-release-date <br> Rating: mock-rating <br> Genre: mock-genre <br> Awards: mock-awards</div>
  </div>
</div>"
`;
